src/main/java/com/github/tomakehurst/wiremock/matching/ContentPatternDeserialiser.java

```diff
@@ -43,20 +42,15 @@ public ContentPattern<?> deserialize(JsonParser parser, DeserializationContext c
     return new StringValuePatternJsonDeserializer().buildStringValuePattern(rootNode);
   }
 
-  private BinaryEqualToPattern deserializeBinaryEqualTo(JsonNode rootNode)
-      throws JsonMappingException {
+  private BinaryEqualToPattern deserializeBinaryEqualTo(JsonNode rootNode) {
     String operand = rootNode.findValue("binaryEqualTo").textValue();
 
     return new BinaryEqualToPattern(operand);
   }
 
   private static boolean isAbsent(JsonNode rootNode) {
-    for (Map.Entry<String, JsonNode> node : ImmutableList.copyOf(rootNode.fields())) {
-      if (node.getKey().equals("absent")) {
-        return true;
-      }
-    }
-
-    return false;
+    List<Map.Entry<String, JsonNode>> list = new LinkedList<>();
+    rootNode.fields().forEachRemaining(list::add);
```

Comment:
Rather than imperatively converting from iterators to something streamable in this fashion, I suggest we adopt a pattern like the one described in:
https://mkyong.com/java8/java-8-how-to-convert-iterator-to-stream/